modules/adding-ibmz-lpar-agent.adoc

```diff
@@ -0,0 +1,105 @@
+[source,yaml]
+----
+rd.neednet=1 cio_ignore=all,!condev \
+console=ttysclp0 \
+ignition.firstboot ignition.platform.id=metal
+coreos.live.rootfs_url=http://<http_server>/rhcos-<version>-live-rootfs.<architecture>.img \// <1>
+coreos.inst.persistent-kargs=console=ttysclp0
+ip=<ip>::<gateway>:<netmask>:<hostname>::none nameserver=<dns> \// <2>
+rd.znet=qeth,<network_adaptor_range>,layer2=1
+rd.<disk_type>=<adapter> \// <3>
+zfcp.allow_lun_scan=0
+ai.ip_cfg_override=1 \// <4>
+random.trust_cpu=on rd.luks.options=discard
+----
+<1> For the `coreos.live.rootfs_url` artifact, specify the matching `rootfs` artifact for the `kernel` and `initramfs` that you are starting. Only HTTP and HTTPS protocols are supported.
+<2> For the `ip` parameter, manually assign the IP address, as described in _Installing a cluster with z/VM on IBM Z and IBM LinuxONE_. 
+<3> For installations on DASD-type disks, use `rd.dasd` to specify the DASD where Red Hat Enterprise Linux CoreOS (RHCOS) is to be installed. For installations on FCP-type disks, use `rd.zfcp=<adapter>,<wwpn>,<lun>` to specify the FCP disk where RHCOS is to be installed.
+<4> Specify this parameter when you use an Open Systems Adapter (OSA) or HiperSockets.
+
+. Generate the `.ins` and `initrd.img.addrsize` files.
++
+The `.ins` file is a special file that includes installation data and is present on the FTP server. It can be accessed from the HMC system.
++
+This file contains details such as mapping of the location of installation data on the disk or FTP server, the memory locations where the data is to be copied.
++
+*Note*: The `.ins` file and `initrd.img.addrsize` are not automatically generated as part of boot-artifacts from the installer. You must manually generate these files.
+
+.. Run the following commands to get the size of the `kernel` and `initrd`: 
++
+[source,terminal]
+----
+KERNEL_IMG_PATH='./kernel.img'
+INITRD_IMG_PATH='./initrd.img'
+CMDLINE_PATH='./generic.prm'
+kernel_size=$(stat -c%s $KERNEL_IMG_PATH )
+initrd_size=$(stat -c%s $INITRD_IMG_PATH)
+----
+.. Round the `kernel` size up to the next MB boundary. This value  is the starting address of `initrd.img`.
++
+[source,terminal]
+----
+offset=$(( (kernel_size + 1048575) / 1048576 * 1048576 ))
+----
+.. Create the kernel binary patch file that contains the `initrd` address and size by running the following commands:
++
+[source,terminal]
+----
+INITRD_IMG_NAME=$(echo $INITRD_IMG_PATH | rev | cut -d '/' -f 1 | rev)
+KERNEL_OFFSET=0x00000000
+KERNEL_CMDLINE_OFFSET=0x00010480
+INITRD_ADDR_SIZE_OFFSET=0x00010408
+OFFSET_HEX=$(printf '0x%08x\n' $offset)
+----
+
+.. Convert the address and size to binary format by running the following commands:
++
+[source,terminal]
+----
+printf "$(printf '%016x\n' $initrd_size)" | xxd -r -p > temp_size.bin
+----
+
+.. Concatenate the address and size binaries by running the following command:
++
+[source,terminal]
+----
+cat temp_address.bin temp_size.bin > "$INITRD_IMG_NAME.addrsize"
```

Comment:
None of these commands start with a $ sign so I assume they are added to a file and run together in a script format?

Reply:
It is a python script, so i will leave it as is.